### PR TITLE
Add testing for Django 5.0 and Wagtail 6.0

### PR DESCRIPTION
**Pull request by nickmoreton**
This is the PR to update the package for Wagtail 6.0 and Django 5.0

It's planned that this will be merged to the main branch and subsequently a PR made to the maintainer.

### CODE REVIEW

File: wagtailsharing/checks.py
Comment (nickmoreton) [inline wagtailsharing/checks.py:1]: Deleted because the app no longer supports Wagtail 5.0

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -37,6 +37,19 @@ jobs:
         python: ['3.8', '3.11']
         django: ['3.2', '4.2']
         wagtail: ['5.1', '5.2']
+        include:
+          - python: '3.12'
+            django: '4.2'
+            wagtail: '5.2'
```
Comment (katdom13) [inline .github/workflows/test.yml:43]: Django 4.2 doesn't support Python 3.12, so this should be removed

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -37,6 +37,19 @@ jobs:
...
+          - python: '3.12'
+            django: '4.2'
+            wagtail: '6.0'
```
Comment (katdom13) [inline .github/workflows/test.yml:46]: This one too

**Comment by nickmoreton** [comment]
> Also wouldn't hurt to run the [django-upgrade](https://github.com/adamchainz/django-upgrade) tool
> `git ls-files -z -- '*.py' | xargs -0 django-upgrade --target-version 5.0`

Thanks @katdom13 that's a good idea 👍

File: .github/workflows/test.yml (hunk 1 above)
Comment (nickmoreton) [reply .github/workflows/test.yml]: Updated in: https://github.com/torchbox-forks/wagtail-sharing/pull/11/commits/346d5234834a566717222a1d3abf6ea33ec3248c

File: .github/workflows/test.yml (hunk 2 above)
Comment (nickmoreton) [reply .github/workflows/test.yml]: Updated in: https://github.com/torchbox-forks/wagtail-sharing/pull/11/commits/346d5234834a566717222a1d3abf6ea33ec3248c

**Comment by nickmoreton** [comment]
> Also wouldn't hurt to run the [django-upgrade](https://github.com/adamchainz/django-upgrade) tool
> `git ls-files -z -- '*.py' | xargs -0 django-upgrade --target-version 5.0`

Might not be possible unless we choose to include conditionals around the changes. I think we'd be moving into an area that could be rejected by the maintainer. I think I'll revert this change for now. What are your thoughts @katdom13 ?

**Comment by nickmoreton** [comment]
Hi @katdom13 For the above the CI testing fails: https://github.com/torchbox-forks/wagtail-sharing/actions/runs/8262291562 so I have reverted using django-upgrade.

**Review by katdom13 — APPROVED** [review]
Thanks @nickmoreton ,
LGTM

**Pull request merged**